src/api/ledger/transaction.js

```diff
@@ -22,29 +168,31 @@ function attachTransactionDate(remote, tx, callback) {
     if (error) {
       callback(new errors.NotFoundError('Transaction ledger not found'));
     } else if (typeof data.ledger.close_time === 'number') {
-      callback(null, _.assign({date: data.ledger.close_time}, tx));
+      callback(undefined, _.assign({date: data.ledger.close_time}, tx));
     } else {
       callback(new errors.ApiError('Ledger missing close_time'));
     }
   });
 }
 
-function isTransactionInRange(tx, options) {
+function isTransactionInRange(tx: Object, options: TransactionOptions) {
   return (!options.minLedgerVersion
           || tx.ledger_index >= options.minLedgerVersion)
       && (!options.maxLedgerVersion
           || tx.ledger_index <= options.maxLedgerVersion);
 }
 
-function getTransactionAsync(identifier, options, callback) {
+function getTransactionAsync(identifier: string, options: TransactionOptions, callback: GetTransactionResponseCallback) {
   validate.identifier(identifier);
   validate.getTransactionOptions(options);
 
   const remote = this.remote;
-  const maxLedgerVersion = Math.min(options.maxLedgerVersion,
```

Comment:
Nice, caught a bug. To avoid using `let`, how about `options.maxLedgerVersion || Infinity`?